src/IECoreArnold/ShaderNetworkAlgo.cpp

```diff
@@ -179,7 +179,10 @@ AtNode *convertWalk( const ShaderNetwork::Parameter &outputParameter, const IECo
 		node = nodeCreator( g_oslArnoldString, AtString( nodeName.c_str() ) );
 		if( oslOutput.string().size() )
 		{
-			AiNodeDeclare( node, g_outputArnoldString, "constant STRING" );
+			if( !AiNodeLookUpUserParameter( node, g_outputArnoldString ) )
```

Comment:
The reason that the parameter has already been declared is that when `ShaderNetworkAlgo::update()` is called to update a light, it re-uses existing nodes wherever possible. It does this using `resetNode()`, so I think the correct fix is to make `resetNode()` remove any user parameters, including the ones added by [`convert()`](https://github.com/GafferHQ/gaffer/blob/main/src/IECoreArnold/ShaderNetworkAlgo.cpp#L381).

Once `resetNode()` has been fixed, it should be possible to use the old code here.